src/features/works/api/worksServices.ts

```diff
@@ -0,0 +1,41 @@
+import { collection, doc, getDoc, setDoc, addDoc, deleteDoc, query, where, getDocs } from 'firebase/firestore';
+import { db } from 'firebaseConfig';
+import { Work } from 'features/works/types';
+
+export const fetchUserWorks = async (userId: string): Promise<Work[]> => {
+    const worksCollection = collection(db, 'works');
+    const q = query(worksCollection, where('userId', '==', userId));
+    const querySnapshot = await getDocs(q);
+    return querySnapshot.docs.map((doc) => ({
+        id: doc.id,
+        ...(doc.data() as Omit<Work, 'id'>),
+        createdAt: doc.data().createdAt.toDate()
+    }));
+};
+
+export const fetchWorkById = async (id: string): Promise<Work | null> => {
+    const docRef = doc(db, 'works', id);
+    const docSnap = await getDoc(docRef);
+    if (docSnap.exists()) {
+        return { id: docSnap.id, ...(docSnap.data() as Omit<Work, 'id'>), createdAt: docSnap.data().createdAt.toDate() };
+    } else {
+        return null;
+    }
+};
+
+export const addWork = async (work: Omit<Work, 'id' | 'createdAt'>): Promise<void> => {
+    await addDoc(collection(db, 'works'), {
+        ...work,
+        createdAt: new Date()
+    });
+};
```

Comment:
```suggestion
export const addWork = (work: Omit<Work, 'id' | 'createdAt'>): Promise<void> => {
   return addDoc(collection(db, 'works'), {
        ...work,
        createdAt: new Date()
    });
};
```
redundant async/await, await it where it's needed

Reply:
If we don't use async we have warning
Promise returned from addDoc is ignored